File: cs50 - x/Python/sentimental-readability/readability.py
from cs50 import get_string
import math

letters = 0
words = 0
sentences = 0

text = get_string("Text: ")

textlen = len(text)

for i in range(textlen):
    if str.isalpha(text[i]):
        letters += 1

    if str.isspace(text[i]):
        words += 1

    if text[i] == '.' or text[i] == '!' or text[i] == '?':
        sentences += 1
words += 1
avg_letters = float(letters)  * 100 // float(words)
avg_sentences = float(sentences)  * 100 // float(words)
grade = round(0.0588 * avg_letters - 0.296 * avg_sentences - 15.8)

if grade >= 1 and grade < 16:
    print(f"Grade {grade}")
else:
    if grade < 1:
        print("Before Grade 1")
    if grade >= 16:
        print("Grade 16+")

File: cs50 - x/Arrays/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string t);
int count_words(string t);
int count_sentences(string t);
int Coleman_index(int l, int w, int s);

int main(void)
{
    string text = get_string("Text: ");
    int letters = count_letters(text);

    printf("number of letters: %i\n", letters);

    int words = count_words(text);

    printf("number of words: %i\n", words);

    int sentences = count_sentences(text);

    printf("number of sentences: %i\n", sentences);

    int grade = Coleman_index(letters, words, sentences);

    if (grade >= 1 && grade < 16)
    {
        printf("Grade %i\n", grade);
    }
    else
    {
        if (grade < 1)
        {
            printf("Before Grade 1\n");
        }
        if (grade >= 16)
        {
            printf("Grade 16+\n");
        }
    }
}




//number of letters function
int count_letters(string t)
{
    int letters = 0;
    int textlen = strlen(t);

    for (int i = 0; i < textlen; i++)
    {
        if (isalpha(t[i]))
        {
            letters++;
        }
    }
    return letters;
}

//number of words function
int count_words(string t)
{
    int words = 0;
    int textlen = strlen(t);

    for (int i = 0; i < textlen; i++)
    {
        if (isspace(t[i]))
        {
            words++;
        }
    }
    words++;
    return words;
}

//number of sentences function
int count_sentences(string t)
{
    int sentences = 0;
    int textlen = strlen(t);

    for (int i = 0; i < textlen; i++)
    {
        if (t[i] == '.' || t[i] == '!' || t[i] == '?')
        {
            sentences++;
        }
    }
    return sentences;
}

//Coleman-Liau index implementation
int Coleman_index(int l, int w, int s)
{
    float al = (float) l / (float) w * 100;                    //average of letters per 100 words
    float as = (float) s / (float) w * 100;                     //average of sentences per 100 words
    float index = 0.0588 * al - 0.296 * as - 15.8;
    int grade = round(index);
    return grade;
}